src/auth/authThunks.js

```diff
@@ -0,0 +1,47 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import TokenGenerator from '../helpers/TokenGenerator';
+import {
+  setAuthToken,
+  setUsername,
+  setError,
+  clearError,
+  logout,
+} from './authSlice';
+
+export const login = createAsyncThunk(
+  'auth/login',
+  async ({ username, password }, { dispatch }) => {
+    try {
+      const { VITE_ADMIN_USERNAME, VITE_ADMIN_PASSWORD } = import.meta.env;
+
+      if (
+        username === VITE_ADMIN_USERNAME &&
+        password === VITE_ADMIN_PASSWORD
+      ) {
+        const authToken = TokenGenerator({ length: 5 });
+        localStorage.setItem('FITNESS_TRACKER_APP_ADMIN_TOKEN', authToken);
+        localStorage.setItem('username', username);
+
+        dispatch(setAuthToken(authToken));
```

Comment:
Pls check how to handle the **thunks** in slice from the link I shared above.

https://redux-toolkit.js.org/api/createAsyncThunk